File: DevExpress-Demo/CS/MVVMDemo.Wpf/Modules/BindableBase/BindablePropertiesViewModel.cs
using DevExpress.Mvvm;

namespace MVVMDemo.BindableBaseDemo {
    public class BindablePropertiesViewModel : BindableBase {
        public string FirstName {
            get { return GetValue<string>(); }
            set { SetValue(value, changedCallback: NotifyFullNameChanged); }
        }

        public string LastName {
            get { return GetValue<string>(); }
            set { SetValue(value, changedCallback: NotifyFullNameChanged); }
        }

        public string FullName { get { return FirstName + " " + LastName; } }

        void NotifyFullNameChanged() {
            RaisePropertyChanged(() => FullName);
        }

    }
}

File: DevExpress-Demo/VB/MVVMDemo.Wpf/Modules/BindableBase/BindablePropertiesViewModel.vb
Imports DevExpress.Mvvm

Namespace MVVMDemo.BindableBaseDemo

    Public Class BindablePropertiesViewModel
        Inherits BindableBase

        Public Property FirstName As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value, changedCallback:=New System.Action(AddressOf NotifyFullNameChanged))
            End Set
        End Property

        Public Property LastName As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value, changedCallback:=New System.Action(AddressOf NotifyFullNameChanged))
            End Set
        End Property

        Public ReadOnly Property FullName As String
            Get
                Return FirstName & " " & LastName
            End Get
        End Property

        Private Sub NotifyFullNameChanged()
            RaisePropertyChanged(Function() FullName)
        End Sub
    End Class
End Namespace
